### PR TITLE
The armoury contraband locker guns have a chance to be rigged.

### PR DESCRIPTION
**Pull request by Pickle-Coding**
## About The Pull Request
Gives a 10% chance for the handguns in the armoury to be rigged. The makarov and desert eagle will have a clown pin, ultra clown pin or the explosive ultra clown pin when rigged, randomly and uniformly weighted. The pins can not be removed, even the normal ones, so there is only one way to find out. The Unica 6 will be reversed, shooting you instead when rigged.
## Why It's Good For The Game
People are too trustworthy of the gear they use in the contraband locker. By giving them a chance to be rigged and backfire on them, people will treat these guns like contraband and not toys. If someone was treating them like a toy, they could receive a nice introduction of the syndicate.
## Changelog
:cl:
balance: There are reports of malfunctioning guns being confiscated.
/:cl:

### CODE REVIEW

**Comment by Helg2** [comment]
Pr is funny, but "rigged" isn't quite informative.

**Comment by SomeoneYouProbablyKnow** [comment]
A glace at the changes shows that it's just the big revolver and it has a 10% chance to be reversed. 

I think that's not a good way of doing this. It's subject to easy checking and metagaming (just the things the type of player to loot armour roundstart is going to do). You don't have to worry about any other gun like the makarov or the shotgun or whatever and as long as you shoot it once you can be certain it won't backfire on you later.

I know it's more work but if you want to do this right the backfire should be a % chance akin to modified det gun and should effect most guns that spawn in the locker. That way testing or metagming this feature is much harder.

**Comment by mc-oofert** [comment]
the contraband locker shouldnt even start with guns anyway

**Comment by Jackal-boop** [comment]
Meh. I don't like this. Easily meta game able and could fuck over antags raiding the armoury

**Comment by Jacquerel** [comment]
I don't know what I would do instead of this but I don't think this solution effectively solves any problems, nor does it really make any sense.
You also failed to actually write a "why is this good for the game section" which is the actual reason I am applying this label: that isn't optional. Tell us what problem you think this _is_ solving and why this is the right solution, because right now I do not see it.

**Comment by Pickle-Coding** [comment]
> A glace at the changes shows that it's just the big revolver and it has a 10% chance to be reversed.

The other guns (except the combat shotgun since that's not necessarily syndicate, but maybe I could add that in since it's in the contraband locker for a reason) can spawn with a clown firing pin. At minimum it would humiliate the user, and worst case scenario for them they get exploded.

> I think that's not a good way of doing this. It's subject to easy checking and metagaming (just the things the type of player to loot armour roundstart is going to do). You don't have to worry about any other gun like the makarov or the shotgun or whatever and as long as you shoot it once you can be certain it won't backfire on you later.
> 
> I know it's more work but if you want to do this right the backfire should be a % chance akin to modified det gun and should effect most guns that spawn in the locker. That way testing or metagming this feature is much harder.

I might look into that.

> I don't know what I would do instead of this but I don't think this solution effectively solves any problems, nor does it really make any sense. You also failed to actually write a "why is this good for the game section" which is the actual reason I am applying this label: that isn't optional. Tell us what problem you think this _is_ solving and why this is the right solution, because right now I do not see it.

Problem: People are too trustworthy of the gear they pick up in the contraband locker. The syndicate managing to trick security to confiscate rigged weapons to backfire them in a future operation sounds like something they would do.

Solution: Rig the weapons in the contraband locker. If someone was too trustworthy of the syndicate gear they are playing with, now they found out. If people are being wary over the syndicate gear in the contraband locker, well now they're treating it like contraband, which is fine.

**Comment by SomeoneYouProbablyKnow** [comment]
> The other guns (except the combat shotgun since that's not necessarily syndicate, but maybe I could add that in since it's in the contraband locker for a reason) can spawn with a clown firing pin. At minimum it would humiliate the user, and worst case scenario for them they get exploded.

Kickass, only like 5 people know how to remove firing pins anyway.

> Problem: People are too trustworthy of the gear they pick up in the contraband locker. The syndicate managing to trick security to confiscate rigged weapons to backfire them in a future operation sounds like something they would do.

Dios mio, a "realism and lore" justification!

**Comment by Pickle-Coding** [comment]
> In the PR body please, rather than halfway down the page 😅 Instead of the two sentences you wrote which neither explain what the feature is nor why we should merge it.

Is this enough?

**Comment by DaydreamIQ** [comment]
This is a funny change but anyone who even remotely knows it exists will just swap out the firing pins anyways. Unless you can somehow make the pin unremovable

**Comment by DaCoolBoss** [comment]
Lame solution to a non existent problem.

**Comment by SomeoneYouProbablyKnow** [comment]
> Lame solution to a non existent problem.
Armoury looting seccie detected, deploy point and laugh squadrons.
But really, it's a funny half measure (People sometimes shoot themselves in the face) for a very minor issue (Security looting that locker for no other reason than they can)

**Comment by SimplySmiley123** [comment]
> > Lame solution to a non existent problem.
> 
> Armoury looting seccie detected, deploy point and laugh squadrons. But really, it's a funny half measure (People sometimes shoot themselves in the face) for a very minor issue (Security looting that locker for no other reason than they can)

Not like they already loot whatever traitor and now also spy gets caught. Often times for something more impactful then a makarov.
All this is is just flavor.

**Comment by Iamgoofball** [comment]
definitely make the pins unremovable

**Comment by Pickle-Coding** [comment]
Done!~ :3

**Comment by moocowswag** [comment]
Good change, security getting a weapon that 1 or 2 shots people due to round start rng is a meme. Now its a bigger gamble

**Comment by Pickle-Coding** [comment]
The pin is always unremovable, even if it's not rigged.

File: code/game/objects/effects/spawners/random/contraband.dm [hunk 1]
```diff
@@ -43,13 +43,14 @@
...
+		/obj/item/gun/ballistic/automatic/pistol/contraband = 8,
 		/obj/item/gun/ballistic/shotgun/automatic/combat = 5,
 		/obj/item/storage/box/syndie_kit/throwing_weapons = 3,
 		/obj/item/grenade/clusterbuster/teargas = 2,
 		/obj/item/grenade/clusterbuster = 2,
-		/obj/item/gun/ballistic/automatic/pistol/deagle,
-		/obj/item/gun/ballistic/revolver/mateba,
+		/obj/item/gun/ballistic/automatic/pistol/deagle/contraband,
+		/obj/item/gun/ballistic/revolver/mateba = 0.9,
+		/obj/item/gun/ballistic/revolver/reverse/mateba = 0.1,
```
Comment (Ghommie) [inline code/game/objects/effects/spawners/random/contraband.dm:53]: I don't think pick_weight() support decimals. If that's the case, multiply every value in the list by 10.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Pull request merged**